addons/account/tests/test_sequence_mixin.py

```diff
@@ -537,3 +554,32 @@ def test_sequence_concurency_no_useless_lock(self):
         # check the values
         moves = env0['account.move'].browse(self.data['move_ids'])
         self.assertEqual(moves.mapped('name'), ['CT/2016/01/0001', '/', 'CT/2016/01/0002'])
+
+    def test_sequence_concurrency_no_deadlock(self):
+        env0, env1, env2 = self.data['envs']
+
+        # start the transactions here on cr1 to simulate concurrency with cr2
+        move = env1['account.move'].browse(self.data['move_ids'][1])
+        move.action_post()
+
+        async def t2():
+            payment = env2['account.payment'].browse(self.data['payment_ids'][1])
+            payment.action_post()
+            move = env2['account.move'].browse(self.data['move_ids'][2])
+            move.action_post()
+            env2.cr.commit()
+        promise = t2()
+
+        asyncio.run(asyncio.sleep(1))  # the first action_post in t2 will never return before this
```

Comment:
That seems like an overly complicated way of writing `time.sleep(1)`.

As to the comment, it certainly won't since it's not running at all.

Reply:
I confirm that it is a complicated way of writing `time.sleep(1)`.